src/scripts/normalize-options.test.ts

```diff
@@ -0,0 +1,52 @@
+import normalizeOptions from "./normalize-options";
+
+jest.mock("process", () => ({
+  cwd: () => "/test/"
+}));
+
+describe("options normalization", () => {
+  it("should handle just source", () => {
+    expect(
+      normalizeOptions({
+        source: "schemas-folder"
+      })
+    ).toEqual({
+      interfaceTarget: "/test/json-schema-interfaces",
+      patterns: {},
+      source: "/test/schemas-folder",
+      validatorTarget: "/test/json-schema-validators"
+    });
+  });
+
+  it("should handle interface target", () => {
+    expect(
+      normalizeOptions({
+        interfaceTarget: "custom-interface-folder",
+        source: "schemas-folder"
+      })
+    ).toEqual({
+      interfaceTarget: "/test/custom-interface-folder",
+      patterns: {},
+      source: "/test/schemas-folder",
+      validatorTarget: "/test/json-schema-validators"
+    });
+  });
+
+  it("should handle validator target", () => {
+    expect(
+      normalizeOptions({
+        validatorTarget: "custom-validator-folder",
+        source: "schemas-folder"
+      })
+    ).toEqual({
+      interfaceTarget: "/test/json-schema-interfaces",
+      patterns: {},
+      source: "/test/schemas-folder",
+      validatorTarget: "/test/custom-validator-folder"
+    });
+  });
+
+  it("should handle patterns", () => {
+    fail("test me!");
```

Comment:
I'm not able to get this piece working using the CLI.
Not entirely following how pattern is supposed to work from the programmatic standpoint either.
Assistance with writing a test for this would be appreciated.

Reply:
Okay will take a look